airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/CdcMssqlSourceAcceptanceTest.java

```diff
@@ -177,4 +187,79 @@ private List<AirbyteStateMessage> filterStateMessages(final List<AirbyteMessage>
         .collect(Collectors.toList());
   }
 
+  @Test
+  protected void testNullValueConversion() throws Exception {
+    final List<ConfiguredAirbyteStream> configuredAirbyteStreams =
+        Lists.newArrayList(new ConfiguredAirbyteStream()
+            .withSyncMode(INCREMENTAL)
+            .withDestinationSyncMode(DestinationSyncMode.APPEND)
+            .withStream(CatalogHelpers.createAirbyteStream(STREAM_NAME3,
+                SCHEMA_NAME,
+                Field.of("id", JsonSchemaType.NUMBER),
+                Field.of("name", JsonSchemaType.STRING),
+                Field.of("userid", JsonSchemaType.NUMBER))
+                .withSourceDefinedCursor(true)
+                .withSourceDefinedPrimaryKey(List.of(List.of("id")))
+                .withSupportedSyncModes(Lists.newArrayList(FULL_REFRESH, INCREMENTAL))));
+
+    final ConfiguredAirbyteCatalog configuredCatalogWithOneStream =
+        new ConfiguredAirbyteCatalog().withStreams(List.of(configuredAirbyteStreams.get(0)));
+
+    final List<AirbyteMessage> airbyteMessages = runRead(configuredCatalogWithOneStream, getState());
+    final List<AirbyteRecordMessage> recordMessages = filterRecords(airbyteMessages);
+    final List<AirbyteStateMessage> stateMessages = airbyteMessages
+        .stream()
+        .filter(m -> m.getType() == AirbyteMessage.Type.STATE)
+        .map(AirbyteMessage::getState)
+        .collect(Collectors.toList());
+    Assert.assertEquals(recordMessages.size(), 1);
+    assertFalse(stateMessages.isEmpty(), "Reason");
+    // TODO validate exact records
+    ObjectMapper mapper = new ObjectMapper();
+
+    assertTrue(cdcFieldsOmitted(recordMessages.get(0).getData()).equals(mapper.readTree("""
+                                                                                        {"id":4, "name":"voyager"}""")));
+
+    // when we run incremental sync again there should be no new records. Run a sync with the latest
+    // state message and assert no records were emitted.
+    JsonNode latestState = null;
+    for (final AirbyteStateMessage stateMessage : stateMessages) {
+      if (stateMessage.getType().equals(AirbyteStateMessage.AirbyteStateType.STREAM)) {
+        latestState = Jsons.jsonNode(stateMessages);
+        break;
+      } else if (stateMessage.getType().equals(AirbyteStateMessage.AirbyteStateType.GLOBAL)) {
+        latestState = Jsons.jsonNode(List.of(Iterables.getLast(stateMessages)));
+        break;
+      } else {
+        throw new RuntimeException("Unknown state type " + stateMessage.getType());
+      }
+    }
```

Comment:
Would it be worth extracting this as a helper function? It's probably been copy-pasted a lot.

Reply:
Done. Put it up in SourceAcceptanceTest